EventBus.Kafka/KafkaEventPublisher.cs

```diff
@@ -0,0 +1,107 @@
+﻿using Confluent.Kafka;
+using Finaps.EventBus.Core.Abstractions;
+using Finaps.EventBus.Kafka.Configuration;
+using Finaps.EventBus.Kafka.Models;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.ObjectPool;
+using System;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+
+namespace EventBus.Kafka
+{
+  public class KafkaEventPublisher : IEventPublisher
+  {
+    // private bool _disposed;
+    private readonly ILogger<KafkaEventPublisher> _logger;
+    private KafkaOptions _options;
+    private ObjectPool<IProducer<int, string>> _channelPool;
+    internal KafkaEventPublisher(
+      ILogger<KafkaEventPublisher> logger,
+      KafkaOptions options
+    )
+    {
+      _logger = logger;
+      _options = options;
+      _channelPool = CreateChannelPool();
+    }
+    
+    public void Publish(string message, string eventName, string messageId)
+    {
+      string topic = GetTopicFromKafkaMessage(message);
+      IProducer<int, string> channel = _channelPool.Get();
+      channel.Produce(topic, CreateMessage(message, eventName));
+      _channelPool.Return(channel);
+    }
+
+    public Task PublishAsync(string message, string eventName, string messageId)
+    {
+      string topic = GetTopicFromKafkaMessage(message);
+      IProducer<int, string> channel = _channelPool.Get();
+      channel.ProduceAsync(topic, CreateMessage(message, eventName));
+      _channelPool.Return(channel);
+      return Task.CompletedTask;
+    }
+
+    public string GetTopicFromKafkaMessage(string message)
+    {
+      var kafkaMessage = JsonSerializer.Deserialize<KafkaMessageEvent>(message);
+      string topic;
+      if(kafkaMessage != null){
+        topic = kafkaMessage.Topic;
+      }
+      else{
+        throw new Exception("Invalid kafka message format");
+      }
+      if(topic == null){
+        throw new Exception("Topic not set in kafka message");
+      }
+      return topic;
+    }
+
+    private Message<int, string> CreateMessage(string message, string eventName)
+    {
+      var headers = new Headers();
+      headers.Add("eventName", Encoding.ASCII.GetBytes(eventName) );
+      var key = new Random().Next(9999);
```

Comment:
consider `guid` ?

Reply:
There is a param, on the publish `messageId` if that is set, you should use that instead.

Reply:
agreed